### PR TITLE
OKTA-242636: build scripts

### PR DESCRIPTION
**Pull request by bryanapellanes-okta**
Added build scripts to build projects and run tests. 
 - build.sh is the default cake shell bootstrapper (identical to the one found in other .Net repos)
- build.ps1 is the default cake powershell bootstrapper (identical to the one found in other .Net repos)
- build.cake is the relevant cake build script

The scripts herein will create the ./artifacts directory containing compiled binaries.  Additional work is necessary to build appropriate nuget packages; this can be handled in subsequent PRs.

Related internal issue for reference: [OKTA-313770](https://oktainc.atlassian.net/browse/OKTA-313770)

### CODE REVIEW

**Comment by bryanapellanes-okta** [comment]
Related internal issue for reference: [OKTA-313770](https://oktainc.atlassian.net/browse/OKTA-313770)

File: build.cake [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+			.SetVerbosity(Verbosity.Minimal));
+	});
+
+Task("Restore-Packages")
+	.Does(() => 
+	{
+		NuGetRestore(solutionFile);
+	});
+
+Task("Build-Android")
```
Comment (laura-rodriguez) [inline build.cake:37]: Should all build tasks depend on `Restore`? For example:
```
Task("Build-Android")
.IsDependentOn("Restore-Packages")
.Does(() =>
// ...
```

File: build.cake [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	{
+		CleanDirectory(artifactsDirectory);
+
+		DotNetBuild(solutionFile, settings => settings
+			.SetConfiguration(configuration)
+			.WithTarget("Clean")
+			.SetVerbosity(Verbosity.Minimal));
+	});
+
+Task("Restore-Packages")
```
Comment (laura-rodriguez) [inline build.cake:31]: Shouldn't this task depend on `clean`?

File: build.cake [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+			settings
+				.SetConfiguration(configuration)   
+				.WithTarget("Build")
+				.WithProperty("Platform", "iPhoneSimulator")
+				.WithProperty("OutputPath", iOSOutputDirectory)
+				.WithProperty("TreatWarningsAsErrors", "false")
+				.SetVerbosity(Verbosity.Minimal));
+	});
+
+Task("Run-Tests")
```
Comment (laura-rodriguez) [inline build.cake:64]: Shouldn't tests tasks depend on build?

File: build.cake [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	.Does(() =>
+	{ 		
+		DotNetBuild(androidProject, settings =>
+			settings
+				.SetConfiguration(configuration)  
+				.WithProperty("OutputPath", androidOutputDirectory)         
+				.WithProperty("DebugSymbols", "false")
+				.WithProperty("TreatWarningsAsErrors", "false")
+				.SetVerbosity(Verbosity.Minimal));
+    });
```
Comment (laura-rodriguez) [inline build.cake:48]: Nit: Fix indentation.

File: build.cake (hunk 2 above)
Comment (bryanapellanes-okta) [reply build.cake:31]: I wanted to treat this as a kind of "atomic" operation.  The build tasks output to the same place, if "Restore-Packages" cleaned then subsequent operations could delete previous output.

File: build.cake (hunk 1 above)
Comment (bryanapellanes-okta) [reply build.cake:37]: I don't understand the question.  Can you clarify?  The build tasks ("Build-Android" and "Build-iOS") each depend on "Restore-Packages".

File: build.cake (hunk 3 above)
Comment (bryanapellanes-okta) [reply build.cake:64]: My local experimentation implies that "DotNetCoreTest" builds prior to running the tests defined in the project.  Dependent projects are also built.  I tested this with the following:

```
./build.sh --target=Clean
./build.sh --target=Run-Tests
```

The tests ran and passed without issue.

File: build.cake (hunk 1 above)
Comment (bryanapellanes-okta) [reply build.cake:37]: Are you suggesting that they should possibly not require restore?

Comment (laura-rodriguez) [reply build.cake:37]: Omg, am I blind? Sorry, I didn't see `IsDependentOn("Restore-Packages")` below.

File: build.cake (hunk 3 above)
Comment (laura-rodriguez) [reply build.cake:64]: Oh yes, you are right. If you want to use the Cake build task instead, you can configure `DotNetCoreTest` to not build the project by doing this:

```
     DotNetCoreTest(string.Format("./src/{0}/{0}.csproj", name), new DotNetCoreTestSettings()
                {
                    Configuration = configuration,
                    NoBuild = true
                });
```

If you still want `DotNetCoreTest` to do that for you, I suggest adding a comment clarifying this.

**Pull request merged**